lib/auth/github.go

```diff
@@ -627,20 +635,30 @@ func (a *Server) validateGithubAuthCallback(ctx context.Context, diagCtx *SSODia
 		return nil, trace.Wrap(err)
 	}
 
-	// Auth was successful, return session, certificate, etc. to caller.
-	auth := authclient.GithubAuthResponse{
-		Req: GithubAuthRequestFromProto(req),
-		Identity: types.ExternalIdentity{
-			ConnectorID: params.ConnectorName,
-			Username:    params.Username,
-		},
-		Username: user.GetName(),
-	}
-
 	// In test flow skip signing and creating web sessions.
 	if req.SSOTestFlow {
 		diagCtx.Info.Success = true
-		return &auth, nil
+		return &authclient.GithubAuthResponse{
+			Req:      GithubAuthRequestFromProto(req),
+			Identity: userResp.makeExternalIdentity(req.ConnectorID),
+			Username: userState.GetName(),
+		}, nil
+	}
+
+	// Auth was successful, return session, certificate, etc. to caller.
+	return a.makeGithubAuthResponse(ctx, req, userState, userResp, params.SessionTTL)
+}
+
+func (a *Server) makeGithubAuthResponse(
```

Comment:
just some refactoring to break the big function. same below to split `getGithubUserAndTeams`